### ISSUE TITLE
Performance issues on macOS

### ISSUE THREAD

**Issue by leereilly** [Zaphyk/project-skylines#3]
This game performs great on my Windows PC, but performance is sluggish on my Macbook Pro*—even with lowest resolution  and lowest quality settings. I recorded some [footage](https://www.youtube.com/watch?v=MFYV_TkLXGg) if that's useful 😄 

<sup>\* 2.8 GHz Intel Core i7, AMD Radeon R9 M370X 2048 MB Intel Iris Pro 1536 MB</sup>

Cheers,
&nbsp;&nbsp;  Lee 🍻 

**PS:** I absolutely love this game!!!

**Comment by Zaphyk** [comment]
What are your Windows PC specs? I don't think the issue is related to macOS necessary just poor optimization, I'll do some test and see if I can find the problem :)

**Comment by leereilly** [comment]
My Windows PC is an aging MSI laptop ~

* Intel Core i7-4710HQ (2.5GHz, 6MB Cache)
* NVIDIA Geforce GTX980M with 8GB GDDR5 VRAM
* 32GB DDR3L 1600 MHz

Let me know if you need a hand testing 😁

**Comment by Zaphyk** [comment]
Actually, I do. Since I don't have a Mac could you run this new version and tell me how the performance goes? [Neon Skylines - Mac x86.zip](https://github.com/Zaphyk/project-skylines/files/1484318/Neon.Skylines.-.Mac.x86.zip)

**Comment by leereilly** [comment]
I definitely see an improvement, but it still jerky/laggy compared to the Windows version. Playing with the lowest quality settings is slightly better than player with the highest. Would it be helpful if I shared a video showing performance?

**Comment by Zaphyk** [comment]
Yes, please do so I can compare it to the first one you showed me.

**Comment by leereilly** [comment]
Roger that - here [it is](https://youtu.be/pbH7Ftu3QXU) starting our with lowest settings and then switching to maximum 🤘

**Comment by Zaphyk** [comment]
I've tried distributing some of the heavy chunk loading across frame which should in theory produce less jerkiness, [here is the new build](https://drive.google.com/open?id=1k-LQWHSnAJJkb7DpgnrUEhSZc4hrWEiu)

**Comment by leereilly** [comment]
`1.02` performance is much, _much_ better - probably up to about 90% compared to how it handles Windows. The graphics are completely different though - not sure if that's intentional? Here's some  [gameplay footage](https://www.youtube.com/watch?v=vde74QhLa94) starting out with lowest quality settings then jumping to highest...

Getting close! 🤘

**Comment by Zaphyk** [comment]
The graphics were definitely not intentional. This last build should be the golden one, it should fix the graphics and mantain the performance improvements. Thank you very much for your help in solving the issue. 😁
[Neon Skylines 1.0.3 - Mac x86.zip](https://github.com/Zaphyk/project-skylines/files/1485242/Neon.Skylines.1.0.3.-.Mac.x86.zip)

**Comment by leereilly** [comment]
You're welcome! Thanks for fixing bugs reported by a random guy on the Internet!!!

The ship now flies super smoothly - at maximum resolution and quality! 🚀 

There's still a little bit of jerkiness/stuttering, but on closer look it looks like the ship position/camera might be changing on the Y-axis. Perhaps as more things are coming into view? 

GIF to illustrate below. The blue line is static - I believe the ship should always be level with it as long as <kbd>W</kbd> and <kbd>S</kbd> haven't been pressed..?

<p align="center">
  <img src="https://user-images.githubusercontent.com/121322/32987434-73dd239e-cca0-11e7-9d0c-c8198ca95576.gif" width="75%">
</p>

**Comment by Zaphyk** [comment]
You are right, it should always be in level with the line. I suspect it's an interpolation issue and I took care of it on commit 25b01015f57065e4098216be364cef6ff63b3022. Here is the latest mac build
[Neon Skylines 1.0.3 - Mac x86.zip](https://github.com/Zaphyk/project-skylines/files/1485896/Neon.Skylines.1.0.3.-.Mac.x86.zip)

**Comment by leereilly** [comment]
Assuming that's the correct `.zip` file above (the previous file was also named `1.0.3`) I don't think I see a noticeable difference at higher resolutions / quality? Here's [some footage](https://www.youtube.com/watch?v=GnRCvco5UmY) showing performance at 640x480 (very low quality), 1600x1200 (medium quality), and 2560x1440 (ultra quality).

**Comment by Zaphyk** [comment]
Yes, It was the correct `.zip` . The build fixed some small jerkiness that was also in the windows build, I thought that was the problem but this to me seems more framerate related/physics timestep. Do you know what is your average framerate while playing? Do you notice big framedrops?

**Comment by leereilly** [comment]
I'm not sure what the best way to give you accurate average framerate information is, but here's some more [video footage](https://www.youtube.com/watch?v=sgGWpBS5kIc) showing gameplay and framerate _windowed_ at 2560x1440 (high quality). Note: the mangled audio at the start is me messing up with the video capture.

Here's the ~short~ GIF version:

<p align="center">
<img src="https://user-images.githubusercontent.com/121322/33004532-2e851762-cd76-11e7-8ba9-fc7f3f0a3847.gif">
</p>


Please let me know if I can do something / provide anything better than that info. Is there some debugging info / logging you can sneak into a build that might be useful?

**Comment by Zaphyk** [comment]
Sorry for the delay, here is a build with some debug info on the screen, it should let me know where the bottleneck is.
[Neon Skylines Debug 1.0 - Mac x86.zip](https://github.com/Zaphyk/project-skylines/files/1504002/Neon.Skylines.Debug.1.0.-.Mac.x86.zip)

**Comment by leereilly** [comment]
Hi @Zaphyk - here's some [footage](https://youtu.be/S37Rw_kw8gY) showing gameplay (i)  2560x1440 @ Ultra and (ii) 600x480 (Medium). Let me know if I can do anything else 🤘

**Comment by Zaphyk** [comment]
Hey, someone suggested of a fix for this. I implemented it and here is the build. Let me know if it works.
[Neon Skylines 1.0.4 - Mac x86.zip](https://github.com/Zaphyk/project-skylines/files/1583879/Neon.Skylines.1.0.4.-.Mac.x86.zip)

**Comment by leereilly** [comment]
Hi - thanks @Zaphyk! I checked it out, but unfortunately I didn't notice any difference between it and the `1.0.3` version from https://github.com/Zaphyk/project-skylines/issues/3#issuecomment-345544072.

**Comment by mike-u** [comment]
I'd like to also report having the same problem on macOS. The game looks good but it's unplayable at 800x600 on Very Low settings.

**Comment by gurix** [comment]
Same story here, I can confirm this on a new macbook pro running high sierra.
